### PR TITLE
Add keywords to control gridline locations.

### PR DESCRIPTION
**Pull request by ajdawson**
This PR allows users to provide locator classes to the `GeoAxes.gridlines()` method to specify the locations gridlines should be drawn at.

Uses an image test, not sure if it is a good idea to add another image test, suggestions welcome.

### CODE REVIEW

File: lib/cartopy/tests/mpl/test_gridliner.py [hunk 1]
```diff
@@ -85,6 +85,16 @@ def test_gridliner():
...
 
 
+@ImageTesting(['gridliner2'])
+def test_gridliner2():
+    ax = plt.axes(projection=ccrs.PlateCarree())
+    xl = mticker.FixedLocator(range(0, 361, 30))
+    yl = mticker.FixedLocator(range(-90, 91, 10))
+    ax.set_global()
+    ax.coastlines()
+    ax.gridlines(xlocator=xl, ylocator=yl)
```
Comment (pelson) [inline lib/cartopy/tests/mpl/test_gridliner.py]: Any chance we could get away without a visual test here? How about checking the gridliner object?

**Comment by pelson** [comment]
:+1: thanks @ajdawson. It would be nice to allow users to pass a list of meridians parallels too - Locators aren't the most user friendly interfaces.

File: lib/cartopy/tests/mpl/test_gridliner.py (hunk 1 above)
Comment (ajdawson) [inline lib/cartopy/tests/mpl/test_gridliner.py]: OK no problem.

**Comment by ajdawson** [comment]
@pelson - I've removed the image test, and replaced with a check on the returned `Gridliner` object. I've also allowed the user to specify an iterable of parallels/meridians in the `GeoAxes.gridlines()` interface, but the `Gridliner` constructor still expects a locator. This seems reasonable to me since it keeps the underlying `Gridliner` class tidy, and a user creating `Gridliner` objects manually is likely to manage dealing with locators also.

File: lib/cartopy/mpl/geoaxes.py [hunk 2]
```diff
@@ -872,6 +874,20 @@ def gridlines(self, crs=None, draw_labels=False, **kwargs):
         * draw_labels
             Label gridlines like axis ticks, around the edge.
 
+        * xlocs
+            An iterable of gridline locations or a
+            :class:`matplotlib.ticker.Locator` instance which will be used to
+            determine the locations of the gridlines in the x-coordinate of
+            the given CRS. Defaults to None, which implies automatic locating
```
Comment (QuLogic) [inline lib/cartopy/mpl/geoaxes.py]: "locating" sounds just a little bit off to me. Maybe "definition" would work?

File: lib/cartopy/mpl/gridliner.py [hunk 3]
```diff
@@ -113,6 +114,18 @@ def __init__(self, axes, crs, draw_labels=False, collection_kwargs=None):
             Toggle whether to draw labels. For finer control, attributes of
             :class:`Gridliner` may be modified individually.
 
+        * xlocator
+            A :class:`matplotlib.ticker.Locator` instance which will be used
+            to determine the locations of the gridlines in the x-coordinate of
+            the given CRS. Defaults to None, which implies automatic locating
```
Comment (QuLogic) [inline lib/cartopy/mpl/gridliner.py]: Same here, of course.

**Comment by pelson** [comment]
Merged with ac51912 onto v0.12.x

File: lib/cartopy/mpl/geoaxes.py (hunk 2 above)
Comment (pelson) [inline lib/cartopy/mpl/geoaxes.py]: I'm comfortable merging the PR with the current terminology, but would be happy if @ajdawson were to submit another PR addressing.

For my 2 cents, the term locating is as good as definition, but it may be that there is another term, or way of re-phrasing which is more clear.
